### DISCUSSION TITLE
Why is SWR so cool?

### DISCUSSION THREAD

**Discussion by rauchg** [vercel/swr#234] [General]
@quietshu I'd love to know!

**Comment by jordn** [comment]
Not a lively discussion, but I 100% agree that it's cool! Wonderful software – thanks @shuding!

**Reply by shuding** [reply]
Thank you! Also shout out to all our awesome contributors!

**Comment by darrenjennings** [comment]
I've often wondered the same. If I had to guess, it's probably because **sw**ea**r**ing makes you think you are cool. Great project!

**Comment by cristianbgp** [comment]
Is this even a discussion? SWR is 100% cool

**Comment by dlemm** [comment]
Discussions should be renamed in "love letters" for this topic <3

**Comment by tkesgar** [comment]
I want to share this story I just realized this week, even though we have been using swr for months.

I was developing this Next.js app using a local back-end. We `useSWR` (heh) to wrap our XHR requests to the back-end. useSWR automatically revalidates on window focus by default, and we did not change that. So I switched to my database GUI to change some values, then I went back to the browser page.

Usually, I just straight hit F5 to refresh the page. But then I noticed that the displayed values have changed. Because of revalidation on focus, useSWR has fetched the new values from the back-end API; no window refresh is needed.

So useSWR allows us to have **live-reload back-end** during development.  That is cool!
